mcodingbot/plugins/peps.py

```diff
@@ -89,19 +127,36 @@ def within_age_cutoff(message_created_at: datetime) -> bool:
     return datetime.now(timezone.utc) - message_created_at <= MAX_AGE_FOR_SEND
 
 
-def get_peps_embed(content: str) -> hikari.Embed | None:
-    pep_refs = [
-        int(ref.group("pep")) for ref in re.finditer(PEP_REGEX, content)
-    ]
-    peps = map(pep_manager.get, sorted(set(pep_refs))[:5])
-    pep_links_message = "\n".join(str(pep) for pep in peps if pep)
+def get_pep_refs(content: hikari.UndefinedNoneOr[str]) -> list[PEPInfo]:
+    """
+    Return a sorted list of all the peps mentioned in a string.
+    """
+    if not content:
+        return []
+
+    peps = sorted(
+        [int(ref.group("pep")) for ref in re.finditer(PEP_REGEX, content)]
+    )
 
-    if not pep_links_message:
-        return None
+    def is_pep(pep: PEPInfo | None) -> TypeGuard[PEPInfo]:
+        return bool(pep)
+
+    return list(filter(is_pep, map(pep_manager.get, peps)))
+
+
+def get_peps_embed(refs: list[PEPInfo]) -> hikari.UndefinedOr[hikari.Embed]:
+    """
+    Get a pep embed from a list of refs. If there are no refs,
+    return `hikari.UNDEFINED`.
+    """
+    if not refs:
+        return hikari.UNDEFINED
+
+    pep_links_message = "\n".join(str(pep) for pep in refs if pep)
```

Comment:
Is the `if pep` at the end of the line needed because the function will always get a list with `PEPInfo`s?
```suggestion
    pep_links_message = "\n".join(str(pep) for pep in refs)
```